### PR TITLE
Rubocop

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
Add rubocop with `TargetRubyVersion: 1.9` + gemfile dependency on rubocop 0.50.0 (the last version that supports 1.9).  Style relies on default rubocop styles for now.

Some minor style fixes from running `rubocop --auto-correct`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-crt/Rakefile [hunk 1]
```diff
@@ -12,15 +13,13 @@ task :compile do
...
-      sh "cmake #{native_dir}"
-    end
+    sh "cmake #{native_dir}" unless File.exist?('CMakeCache.txt')
     sh "cmake --build #{build_dir}"
   end
 end
 
 desc 'Move the compiled lib into bin'
-task :bin => :compile do
+task bin: :compile do
```
Comment (graebm) [inline aws-crt/Rakefile:22]: trivial: I have to say, I liked it better with the arrows
can we ignore that rule in this file?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,7 +15,6 @@ env:
   RUN: ${{ github.run_id }}-${{ github.run_number }}
 
 jobs:
-
```
Comment (graebm) [inline .github/workflows/ci.yml:18]: We should add a Github action to run rubocop!
Here's one I found: https://github.com/marketplace/actions/rubocop-linter-action
We could do this in a followup CR

File: aws-crt/Rakefile (hunk 1 above)
Comment (alextwoods) [reply aws-crt/Rakefile]: This style has been proposed a few times, eg: https://github.com/rubocop-hq/rubocop/issues/1437 and https://github.com/rubocop-hq/rubocop/issues/519 but hasn't really gained support.  

I agree I prefer the `=>` (aka hash rocket) for defining tasks as I find it more readable.  I'll disable this check for just the Rakefile.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alextwoods) [reply .github/workflows/ci.yml]: Agreed - I'd like to add that as a check. There are currently a few rubocop violations that couldn't be auto-corrected.  I was waiting for you're refactor to move things around before fixing them + adding a github action.

**Pull request merged**